### ISSUE TITLE
Training grobidTagger without embeddings

### ISSUE THREAD

**Issue by de-code** [kermitt2/delft#33]
Hi @kermitt2, should training without embedding work?

From the readme:

> Reduce model size, in particular by removing word embeddings from them. For instance, the model for the toxic comment classifier went down from a size of 230 MB with embeddings to 1.8 MB. In practice the size of all the models of DeLFT is less than 2 MB, except for Ontonotes 5.0 NER model which is 4.7 MB.

When I try to set `embeddings_name` to `None`, it falls over soon after. I tried to fix the next two issues but there are more. Which makes me think maybe it's not meant to work?

**Comment by kermitt2** [comment]
Hello @de-code !

Yes all these models required embeddings for training and prediction. You can use `word2vec-pmc` for instance with the life science domain. 

The quote refers to the fact that usual keras models store the embeddings weights in the model - thus leading super fat models, while it's useless - and in addition they require the embeddings in memory to be passed to the embeddings layer for prediction (to cover out of vocabulary from what is known by the training) - and that's several GB in memory again lost.
In practice for serious ML NLP applications we need several models and several embeddings to be used at the same time so it's blocking.

The solution in DeLFT is first to separate the embeddings from the model, and have only embeddings injected via a generator, and second have any amount of embeddings in a embedded database which has minimal memory footprint and negligible runtime impact.

**Comment by de-code** [comment]
Okay, would maybe be good if no embedding would also be an option.

Where can I get `word2vec-pmc` from?

**Comment by de-code** [comment]
> Where can I get `word2vec-pmc` from?

Is it this one? http://evexdb.org/pmresources/vec-space-models/wikipedia-pubmed-and-PMC-w2v.bin

**Comment by kermitt2** [comment]
yes this one! I need to add it in the doc.

**Comment by kermitt2** [comment]
Ah I forgot, you need to convert the w2v binary format into the standard .vec before loading it into DeLFT, this cab be done for instance via https://github.com/marekrei/convertvec
(I might have a version in .vec somewhere on S3 but I don't find it anymore)

**Comment by lfoppiano** [comment]
Regarding this issue, do we want to have the possibility to disable the embeddings at all? If not we should close this issue.

**Comment by kermitt2** [comment]
We always need embeddings, but we could of course create the embeddings from the words and word distribution in the training data. Doing so however gives significantly lower accuracy results (lower than simple CRF), I think there is no point not using pre-trained embeddings with this kind of architectures.
